include/flyft/parameter.h

```diff
@@ -132,7 +132,77 @@ class LinearParameter : public DoubleParameter
         double time_;
         double value_;
     };
+    
+class SquareRootParameter : public DoubleParameter
+    {
+    public:
+        SquareRootParameter(double initial, double origin, double rate)
+            : initial_(initial), origin_(origin), rate_(rate), time_(std::nan("")), value_(std::nan(""))
+            {
+            }
 
-}
+        double operator()(std::shared_ptr<State> state) override
+            {
+            // record time for evaluation
+            const double time = state->getTime();
+            if (time != time_)
+                {
+                time_ = time;
+                this->token_.stage();
+                }
+
+            // if anything has changed, evaluate and store
+            if (this->token_.dirty())
+                {
+                if(time_ < 0)
+                    {
+                    throw std::invalid_argument("Complex numbers not allowed");
+                    }
+                value_ = sqrt(initial_*initial_+rate_*(time_-origin_));
```

Comment:
The issue isn't actually if the time is negative, it's if the argument of the square root is negative. You should do like this:
```suggestion
                const double value_sq = initial_*initial_+rate_*(time_-origin_);
                if(value_sq < 0)
                    {
                    throw std::invalid_argument("Time makes argument of square root negative");
                    }
                value_ = std::sqrt(value_sq);
```
Note I'm also using `std::sqrt`.